### PR TITLE
[tests][xtro] Add enum member/values checks. Fix #7527

### LINKED ISSUE #7527: Add enum member/values to Xtro checks
Xtro checks for missing types, methods... but currently does not check if all enum members are included.

ref: https://github.com/xamarin/xamarin-macios/pull/6781

> The most important thing for enum members is their values,
> e.g. if we have MissingMember = 43 while the right value is 42. 
> That's very hard to diagnose such errors (and easy to get wrong
>  since, most of the time, the values are assigned automatically 
> by the C# compiler). It's also a breaking change to fix them.

### PR DESCRIPTION
**Pull request by spouliot**
Figure out if
* we're missing enum values (easy to workaround, but annoying for developers)
* we expose enum values that are not defined natively (potential bugs)

references:
* https://github.com/xamarin/xamarin-macios/issues/5843
* https://github.com/xamarin/xamarin-macios/issues/7527

# Breaking Changes

Since the integral value is baked into the binary and no fields are removed then it won't affect source builds and _minimally_ binaries (need to be rebuilt to pick up the right value).

* `ARKit.ARFrameSemantics.SceneDepth` value
* `AudioToolbox.SmpteTimeType` all values were off by one
* `CoreNFC.NFCReaderError.NdefReaderSessionError*` values
* `HealthKit.HKErrorCode.NoData` value
* `Metal.MTLCounterSampleBufferError.Internal` value
* `MetalPerformanceShaders.MPSCnnNeuronType.Count` value (should never be used as it will change when new values are added)
* `UIKit.UIRectCorner.AllCorners` value (larger mask)
* `UIKit.UIUserInterfaceIdiom.Mac` value (off by one)

note: some will be XM only - but packages / API diff are disabled right now.

### CODE REVIEW

File: tests/xtro-sharpie/EnumCheck.cs [hunk 1]
```diff
@@ -122,25 +126,107 @@ public override void VisitEnumDecl (EnumDecl decl, VisitKind visitKind)
...
 				break;
 			default:
 				throw new NotImplementedException ();
 			}
+
+			var fields = type.Fields;
+			if (signed) {
+				managed_signed_values.Clear ();
+				native_signed_values.Clear ();
+				// skip `0` since it's `__value` (special)
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/EnumCheck.cs:165]: Is the `__value` field guaranteed to always be the first field?

Comment (spouliot) [reply tests/xtro-sharpie/EnumCheck.cs:165]: It's actually `value__` and there's no order mentioning in the ECMA spec
I guess it's easier for compiler to emit the underlying size before adding members :)
It's easy to fix (along with the typo)

**Review by whitneyschmidt — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: tests/xtro-sharpie/watchOS-SceneKit.ignore [hunk 2]
```diff
@@ -17,3 +17,7 @@
 !missing-null-allowed! 'SceneKit.SCNShape SceneKit.SCNShape::Create(UIKit.UIBezierPath,System.nfloat)' is missing an [NullAllowed] on parameter #0
 !missing-null-allowed! 'System.Void SceneKit.SCNText::set_ChamferProfile(UIKit.UIBezierPath)' is missing an [NullAllowed] on parameter #0
 !missing-null-allowed! 'System.Void SceneKit.SCNText::set_Font(UIKit.UIFont)' is missing an [NullAllowed] on parameter #0
+
+# watchOS is 32bits but our [Native] enums are all 64 bits
+!extra-enum-value! Managed value 18446744073709551615 for SCNPhysicsCollisionCategory.All not found in native headers
+!missing-enum-value! SCNPhysicsCollisionCategory native value SCNPhysicsCollisionCategoryAll = 4294967295 not bound
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/watchOS-SceneKit.ignore:23]: Maybe add an exception to the enum check that int.MaxValue and long.MaxValue are acceptable synonyms for [Native] enums?

**Review by rolfbjarne — APPROVED** [review]

File: tests/xtro-sharpie/watchOS-SceneKit.ignore (hunk 2 above)
Comment (spouliot) [reply tests/xtro-sharpie/watchOS-SceneKit.ignore:23]: not sure it's safe to assume 64 == 32, but it should be fine to make 32 == 64

**Review by dalexsoto — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by spouliot** [comment]
@monojenkins backport to xcode12.2

**Comment by spouliot** [comment]
@monojenkins backport to xcode12.2

**Pull request merged**